src/js/ripple/orderbook.js

```diff
@@ -439,19 +439,22 @@ OrderBook.prototype.setFundedAmount = function(offer, fundedAmount) {
     return offer;
   }
 
-  var isOfferGetsExceeded = Amount.from_json(fundedAmount)
+  var isOfferGetsExceeded = Amount.from_json(
+    this._currencyGets.is_native()
+    ? fundedAmount
+    : fundedAmount + OrderBook.IOU_SUFFIX
```

Comment:
It's not a really ugly hack, and there is at least one case where your solution wouldn't work. This function and the rest of the orderbook code could use some refactoring, but that's not part of this bug fix. Thanks for the comments